{
  "source":"print 'globals:', dir()\nprint '__name__:', __name__\n\ndef func(y):\n    def inner(z):\n       return z+2\n    print 'inner returned', inner(3)\n \n    raise NameError('asdf')\n\n    return y+10\n\nif __name__ == '__main__':\n\n    x = 2\n    print 'x is', x\n    x += 3\n    print 'x is now', x\n    try:\n        x = func(x)\n    except NameError as e:\n        print e\n        print 'x is now now', x\n\n    a = []\n    a.append(5)\n    a.append(1)\n    a.append(2)\n    print a\n",
  "trace":[
    {
      "line":0,
      "stdout":"globals: ['__builtins__', '__name__']\n"
    },
    {
      "line":1,
      "stdout":"__name__: __main__\n"
    },
    {
      "line":3
    },
    {
      "line":12
    },
    {
      "line":14
    },
    {
      "line":15,
      "stdout":"x is 2\n"
    },
    {
      "line":16
    },
    {
      "line":17,
      "stdout":"x is now 5\n"
    },
    {
      "line":18
    },
    {
      "line":19
    },
    {
      "line":3
    },
    {
      "line":4
    },
    {
      "line":6,
      "stdout":"inner returned"
    },
    {
      "line":4
    },
    {
      "line":5
    },
    {
      "line":5,
      "stdout":" 5\n"
    },
    {
      "line":8
    },
    {
      "line":20
    },
    {
      "line":21,
      "stdout":"asdf\n"
    },
    {
      "line":22,
      "stdout":"x is now now 5\n"
    },
    {
      "line":24
    },
    {
      "line":25
    },
    {
      "line":26
    },
    {
      "line":27
    },
    {
      "line":28,
      "stdout":"[5, 1, 2]\n"
    }
  ]
}
